shared/chat/conversation/input-area/normal/mention-input.js

```diff
@@ -219,27 +243,33 @@ class MentionInput extends React.Component<MentionInputProps, MentionState> {
 
   // End desktop only.
 
-  _forceSubmit = () => {
+  // If you type @notausername<enter> or #notachannel<enter>, we've essentially 'picked' nothing and really want to submit ... all the way out.
+  //
+  // If you type e.g., @notausername<tab> we don't want to do anything.
+  _maybeSubmitAfterInsert = (notUserOrChannel: boolean) => {
     const text = this._getText()
-    if (text) {
+    if (text && notUserOrChannel && this._lastSelectionKey === 'Enter') {
+      this._setLastSelectionKey('')
```

Comment:
Probably want to call `_setLastSelectionKey('')` unconditonally, so:

```
const text = ...
if (text ...) {
  this.props.onSubmit(text)
}
this._setLastSelectionKey('')
```

Reply:
I recall resetting `_lastSelectionKey` unconditionally and getting some strange behavior, but testing it now it seems to be fine.